.github/workflows/update-lock-files.yml

```diff
@@ -25,9 +25,8 @@ jobs:
           - name: free-threaded
             update_script_args: "--select-tag free-threaded"
             additional_commit_message: "[free-threaded]"
-          - name: cirrus-arm
+          - name: arm
```

Comment:
My understanding is that currently arm is run on every push to a PR. If that's the case, we can remove this section and change the tag in `update_environment_and_lock_files.py` to be `main-ci`

Reply:
That is correct. Applied your suggestion in [`c198487` (#30797)](https://github.com/scikit-learn/scikit-learn/pull/30797/commits/c198487d42952baaea4ea17baa6f92daa8853aec)